codalab/rest/worksheet_block_schemas.py

```diff
@@ -0,0 +1,105 @@
+"""
+Marshmallow schemas that represent worksheet block.
+Used for serializing resource dicts into JSON API documents, and vice-versa.
+The schemas also perform some basic validation.
+"""
+from bottle import local
+from marshmallow import (
+    Schema as PlainSchema,
+    ValidationError,
+    validate,
+    validates_schema,
+)
+from marshmallow_jsonapi import Schema, fields
+import sys
+
+# Enum that represents different modes for a block.
+class BlockModes:
+    markup_block = 'markup_block'
+    record_block = 'record_block'
+    table_block = 'table_block'
+    contents_block = 'contents_block'
+    html_block = 'html_block'
+    image_block = 'image_block'
+    graph_block = 'graph_block'
+    wsearch_block = 'wsearch_block'
+    search_block = 'search_block'
```

Comment:
I think we decided that `search` was not going to be a separate type of block, but a mechanism for retrieving a list of bundles; specifying a list of bundles verbatim is another mechanism.